cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/NamespaceExistenceVerifier.java

```diff
@@ -41,9 +41,7 @@
   public void ensureExists(NamespaceId namespaceId) throws NotFoundException {
     if (!NamespaceId.SYSTEM.equals(namespaceId)) {
       try {
-        if (namespaceQueryAdmin.get(namespaceId.toId()) == null) {
-          throw new NamespaceNotFoundException(namespaceId.toId());
-        }
+        namespaceQueryAdmin.exists(namespaceId.toId());
```

Comment:
We do have NamespaceQueryAdmin the interface we just moved the implementation in DefaultNamespaceAdmin. Since here we are just querying we are  working with QueryAdmin interface.